File: insertionSort.js
const insert = (array) => {
  var i;
  var j;
  for (let i = 1; i < array.length; i++) {
    j = i;
    while (j > 0 && array[j - 1] > array[j]) {
      // swap array[j] and array[j - 1]
      var x = array[j];
      array[j] = array[j - 1];
      array[j - 1] = x;
    } // end while  
  } // end for
  return array;
}

insert([4, 1, 3, 9, 7])

File: insertion_sort.py
def insert(list):
    # i = 1
    # j = i
    for i in range(1, len(list)):
        j = i
        while i < len(list):
            while (j > 0 and list[j - 1] > list[j]):
                # swap list[j] and list[j - 1]
                x = list[j]
                list[j] = list[j - 1]
                list[j - 1] = x
        i = i + 1

    return list

insert([4, 1, 3, 9, 7])
